test/configuration_test.dart

```diff
@@ -527,7 +527,29 @@ Future<void> main([List<String>? args]) async {
     realm.close();
 
     final disconnectedSyncConfig = Configuration.disconnectedSync(schema, path: realmPath);
-    final disconnectedRealm = Realm(disconnectedSyncConfig);
+    final disconnectedRealm = getRealm(disconnectedSyncConfig);
     expect(disconnectedRealm.find<Task>(oid), isNotNull);
   });
+
+  test('Configuration set short encryption key', () {
```

Comment:
What would happen if I create an encryption key containing numbers that don't fit into a byte? E.g. `List<int>.generate(encryptionKeySize, (i) => random.nextInt(1024))`?

Reply:
It is a good point. I added some tests with generated list with exceeding byte size items. It seems to be working. I suppose they are truncated and you are even able to open realm twice with the same key. I'm not sure whether we to throw an exception or to leave it like this.

Reply:
Hm... interesting. I think we may want to validate it to make sure people are not supplying nonsensical data, but perhaps I'm being overly cautious.

Reply:
I think it is not a good idea to add `Uint8List` in the external API, since it requires `typed_data.dart`.

Reply:
Oh, I didn't mean that - more like, in the configuration constructor, do something like:

```dart
  Configuration._(
    this.schemaObjects, {
    String? path,
    this.fifoFilesFallbackPath,
    this.encryptionKey,
  }) {
    _validateEncryptionKey(encryptionKey);

    this.path = path ?? _path.join(_path.dirname(_defaultPath), _path.basename(defaultRealmName));
  }

  void _validateEncryptionKey(List<int>? key) {
    if (key == null) {
      return;
    }

    if (key.length != realmCore.encryptionKeySize) {
      throw RealmException("Wrong encryption key size (must be ${realmCore.encryptionKeySize}, but was ${key.length})");
    }

    final notAByteElement = key.firstWhereOrNull((e) => e > 255);
    if (notAByteElement != null) {
      throw RealmException("Encryption key must be a list of bytes, but found $notAByteElement at index ${key.indexOf(notAByteElement)}");
    }
  }
```

Reply:
Done.